packages/tools/kadena-cli/src/keys/commands/keysDelete.ts

```diff
@@ -0,0 +1,101 @@
+import type { Command } from 'commander';
+import { Option } from 'commander';
+import { z } from 'zod';
+
+import { PLAIN_KEY_DIR } from '../../constants/config.js';
+import { services } from '../../services/index.js';
+import type { CommandResult } from '../../utils/command.util.js';
+import { assertCommandError } from '../../utils/command.util.js';
+import { createCommand } from '../../utils/createCommand.js';
+import { createOption } from '../../utils/createOption.js';
+import { log } from '../../utils/logger.js';
+import { select } from '../../utils/prompts.js';
+import { keysOptions } from '../keysOptions.js';
+
+export const deleteKey = async (key: string): Promise<CommandResult<{}>> => {
+  try {
+    await services.filesystem.deleteFile(`${PLAIN_KEY_DIR}/${key}`);
+    return { success: true, data: {} };
+  } catch (error) {
+    return { success: false, errors: ['Failed to delete key'] };
+  }
+};
+
+export const deleteAllKeys = async (): Promise<CommandResult<{}>> => {
+  try {
+    await services.filesystem.deleteDirectory(PLAIN_KEY_DIR);
+    return { success: true, data: {} };
+  } catch (error) {
+    return { success: false, errors: ['Failed to delete all keys'] };
+  }
+};
+
+const confirmDelete = createOption({
+  key: 'confirm',
+  defaultIsOptional: false,
+  async prompt(args) {
+    if (typeof args.keyFiles !== 'string') return false;
+
+    if (args.keyAlias === 'all') {
+      return await select({
+        message: 'Are you sure you want to delete ALL keys',
+        choices: [
+          { value: true, name: 'Yes, delete all keys' },
+          { value: false, name: 'No, do not delete any keys' },
+        ],
+      });
+    }
+
+    const keyText =
+      args.keyAlias === 'all'
+        ? 'Are you sure you want to delete ALL keys'
+        : `Are you sure you want to delete the key: "${args.keyFiles}"`;
+
+    return await select({
+      message: keyText,
+      choices: [
+        { value: true, name: 'Yes' },
+        { value: false, name: 'No' },
+      ],
+    });
+  },
+  validation: z.boolean(),
+  option: new Option('--confirm', 'Confirm key deletion'),
+});
+
+export const createDeleteKeysCommand: (
+  program: Command,
+  version: string,
+) => void = createCommand(
+  'delete',
+  'Delete key(s) from your local filesystem',
+  [keysOptions.keysFiles(), confirmDelete()],
+  async (option, { collect }) => {
+    const config = await collect(option);
+    if (config.confirm !== true) {
+      log.warning('\nNo key(s) were deleted.\n');
+      return;
+    }
+
+    log.debug('delete-keys:action', { config });
+    if (config.keyFiles === 'all') {
+      const result = await deleteAllKeys();
+      assertCommandError(result);
+      log.info(log.color.green('\nAll keys have been deleted.\n'));
+    } else {
+      if (config.keyFiles === null) {
+        throw new Error(`Key: ${config.keyFiles} does not exist.`);
```

Comment:
```suggestion
        throw new Error(`Key: ${config.keyFiles} is invalid`);
```

Reply:
why is invalid?

Reply:
here the input is invalid because it's null.